beater/server.go

```diff
@@ -18,103 +18,109 @@
 package beater
 
 import (
-	"context"
+	"bufio"
+	"fmt"
 	"net"
 	"net/http"
+	"time"
 
+	"golang.org/x/net/context"
+
+	"github.com/elastic/beats/libbeat/beat"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/version"
 	"go.elastic.co/apm"
-	"go.elastic.co/apm/module/apmhttp"
-	"golang.org/x/net/netutil"
 
-	"github.com/elastic/apm-server/beater/api"
 	"github.com/elastic/apm-server/beater/config"
 	"github.com/elastic/apm-server/publish"
-	"github.com/elastic/beats/libbeat/common/transport/tlscommon"
-	"github.com/elastic/beats/libbeat/logp"
-	"github.com/elastic/beats/libbeat/version"
 )
 
-func newServer(cfg *config.Config, tracer *apm.Tracer, report publish.Reporter) (*http.Server, error) {
-	mux, err := api.NewMux(cfg, report)
+type server struct {
+	logger *logp.Logger
+	cfg    *config.Config
+
+	httpServer   *httpServer
+	otCollectors *otcollector
+}
+
+func newServer(logger *logp.Logger, cfg *config.Config, tracer *apm.Tracer, reporter publish.Reporter) (server, error) {
+	httpServer, err := newHTTPServer(logger, cfg, tracer, reporter)
+	if err != nil {
+		return server{}, err
+	}
+	grpc, err := newOTCollectors(logger, cfg, tracer, reporter)
```

Comment:
s/grpc/otcollectors/ ?

Reply:
BTW many folks use "otel" as the abbreviation, to avoid the naming clash with ot-as-in-opentracing.